src/script/lua_api/l_localplayer.cpp

```diff
@@ -422,6 +422,25 @@ int LuaLocalPlayer::l_hud_get(lua_State *L)
 	return 1;
 }
 
+// hud_get_elements(self)
+int LuaLocalPlayer::l_hud_get_elements(lua_State *L)
+{
+	LocalPlayer *player = getobject(L, 1);
+	if (player == nullptr)
+		return 0;
+
+	lua_newtable(L);
+	u32 max = player->getHudIdMax();
+	for (u32 id = 0; id < max; id++) {
+		HudElement *elem = player->getHud(id);
```

Comment:
Please get the entire list of active HUD IDs from the server. This loop is very inefficient.

Reply:
fixed

Reply:
I did this, because I was unsure about the Mutex in the Player class.
Now it use a function pointer, hopefully it is faster now.

Reply:
Now it looks uglier but is definitely faster.